util/cmd/cmd.go

```diff
@@ -45,6 +45,13 @@ func PrintVersion(cliName string, version wfv1.Version, short bool) {
 	fmt.Printf("  Platform: %s\n", version.Platform)
 }
 
+// PrintVersionMismatchWarning detects if there's a mismatch between the client and server versions and prints a warning if so
+func PrintVersionMismatchWarning(clientVersion wfv1.Version, serverVersion string) {
+	if serverVersion != "" && clientVersion.GitTag != "" && serverVersion != clientVersion.Version {
```

Comment:
this is identical to the previous PR, although I wonder if we want to have some sort of behavior for unset versions. 🤔
that would normally only happen with development versions and a contributor would be aware of this, but I'm thinking in the case of a untagged internal build or something, a warning could be useful. although that's pretty niche of a case, so probably not worthwhile

Reply:
That could be useful, but I'm worried printing a warning every time an untagged build of the Argo CLI is run will become more annoying than helpful. We could add a CLI flag to suppress the warning, but that seems overkill.